gui/include/streamsession.h

```diff
@@ -47,6 +48,7 @@ class ChiakiException: public Exception
 
 struct StreamSessionConnectInfo
 {
+ 	Settings * settings;
```

Comment:
This struct shouldn't depend on the actual `Settings` class. Better add a member of type `QMap<Qt::Key, int>` and extract the value in the `StreamSessionConnectInfo(Settings *settings, QString host, QByteArray regist_key, QByteArray morning)` constructor.

Reply:
How do you feel about passing `Settings *` to `StreamWindow` constructor, in order to let it connect to a signal in `Setttings` that would be called `KeyboardMapChanged`? That would enable the user to change the mapping without having to reopen the stream.

Reply:
or I can do what you say, but then at the same time connect the slot externally to StreamWindow. I think that would be better.

Reply:
humm.. I realized that the `StreamWindow` is consuming all keypress, and not forwarding to other windows, so changing the key mapping while stream is on is not possible. Not sure if want to change this behavior.